core/src/main/java/org/elasticsearch/index/mapper/internal/SeqNoFieldMapper.java

```diff
@@ -108,6 +114,34 @@ public Query termQuery(Object value, @Nullable QueryShardContext context) {
             throw new QueryShardException(context, "SeqNoField field [" + name() + "] is not searchable");
         }
 
+        @Override
+        public FieldStats stats(IndexReader reader) throws IOException {
+            final List<LeafReaderContext> leaves = reader.leaves();
+            if (leaves.isEmpty()) {
+                return null;
+            }
+
+            long currentMin = Long.MAX_VALUE;
+            long currentMax = Long.MIN_VALUE;
+            boolean found = false;
+            for (int i = 0; i < leaves.size(); i++) {
+                final LeafReader leaf = leaves.get(i).reader();
+                final NumericDocValues values = leaf.getNumericDocValues(name());
+                if (values == null) continue;
+                found = true;
```

Comment:
strictly speaking found needs to be in the inner loop - if all docs are deleted, we didn't find anything.

PS. That said for what we don't need to care about deletes in the current implementation - if fact, we can remove all the live docs logic (although it's strictly  speaking correct from a stats perspective). I'm OK with removing live docs  completely  (instead of inlining the found or relying on min and max staying at their initial value).